### PR TITLE
Refactor core Gradle Metadata attributes

### PR DESCRIPTION
**Pull request by ljacomet**
See #9675 for details

### CODE REVIEW

**Comment by ljacomet** [comment]
Bump to ready for review was a click mistake, consider WIP still.

File: subprojects/model-core/src/main/java/org/gradle/internal/snapshot/impl/AbstractIsolatableScalarValue.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 package org.gradle.internal.snapshot.impl;
 
-import org.gradle.internal.isolation.Isolatable;
+import org.gradle.internal.isolation .Isolatable;
```
Comment (melix) [inline subprojects/model-core/src/main/java/org/gradle/internal/snapshot/impl/AbstractIsolatableScalarValue.java]: :o: This is probably not intentional

File: subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/metadata/GradleModuleMetadataCompatibilityConverter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import org.gradle.api.internal.attributes.ImmutableAttributes;
+import org.gradle.api.internal.attributes.ImmutableAttributesFactory;
+import org.gradle.api.internal.model.NamedObjectInstantiator;
+import org.gradle.internal.component.external.model.MutableComponentVariant;
+import org.gradle.internal.component.external.model.MutableModuleComponentResolveMetadata;
+import org.gradle.internal.snapshot.impl.CoercingStringValueSnapshot;
+
+public class GradleModuleMetadataCompatibilityConverter {
+
+    public static final Attribute<String> USAGE_STING_ATTRIBUTE = Attribute.of(Usage.USAGE_ATTRIBUTE.getName(), String.class);
```
Comment (melix) [inline subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/metadata/GradleModuleMetadataCompatibilityConverter.java]: :o: Call the police! Typo :dancing_men:

Comment (melix) [reply subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/metadata/GradleModuleMetadataCompatibilityConverter.java]: :x:  those constants should be private and `FORMAT_STRING_ATTRIBUTE` probably deserves a rename.

File: subprojects/core-api/src/main/java/org/gradle/api/attributes/Category.java [hunk 3]
```diff
@@ -53,4 +54,12 @@
      * The enforced platform, usually a synthetic variant derived from the {@code platform}
      */
     String ENFORCED_PLATFORM = "enforced-platform";
```
Comment (andreikokarev) [inline subprojects/core-api/src/main/java/org/gradle/api/attributes/Category.java]: - [ ] 

- 
Duplicate of #

**Pull request merged**